### PR TITLE
Add shopping cart model

### PR DESCRIPTION
**Pull request by RomanSoproniuk**

### CODE REVIEW

File: src/main/java/book/store/onlinebookstore/mapper/ShoppingCartMapper.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import book.store.onlinebookstore.model.ShoppingCart;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(config = MapperConfig.class, uses = UserMapper.class)
+public interface ShoppingCartMapper {
+
+    @Mapping(target = "usersId", source = "user", qualifiedByName = "getUserId")
+    ShoppingCartResponseDto toDto(ShoppingCart cart);
+
```
Comment (okuzan) [inline src/main/java/book/store/onlinebookstore/mapper/ShoppingCartMapper.java:14]:
```suggestion
    @Mapping(target = "usersId", source = "user", qualifiedByName = "getUserId")
    ShoppingCartResponseDto toDto(ShoppingCart cart);
```

File: src/main/java/book/store/onlinebookstore/model/CartItem.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import jakarta.persistence.OneToOne;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity(name = "cart_items")
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE cart_items SET is_deleted = true WHERE id=?")
```
Comment (okuzan) [inline src/main/java/book/store/onlinebookstore/model/CartItem.java:20]:
```suggestion
@SQLDelete(sql = "UPDATE cart_items SET is_deleted = TRUE WHERE id = ?")
```

File: src/main/java/book/store/onlinebookstore/model/ShoppingCart.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity(name = "shopping_carts")
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = true WHERE id=?")
```
Comment (okuzan) [inline src/main/java/book/store/onlinebookstore/model/ShoppingCart.java:22]:
```suggestion
@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = TRUE WHERE id = ?")
```

File: src/main/java/book/store/onlinebookstore/repository/shoppingcart/ShoppingCartRepository.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package book.store.onlinebookstore.repository.shoppingcart;
+
+import book.store.onlinebookstore.model.ShoppingCart;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+
```
Comment (okuzan) [inline src/main/java/book/store/onlinebookstore/repository/shoppingcart/ShoppingCartRepository.java:9]:
```suggestion
```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 5]
```diff
@@ -17,3 +17,9 @@ databaseChangeLog:
       file: db/changelog/changes/008-create-categories-table.yaml
   - include:
       file: db/changelog/changes/009-create-books-categories-table.yaml
+  - include:
+      file: db/changelog/changes/010-create-shopping-carts-table.yaml
+  - include:
+      file: db/changelog/changes/011-add-shopping-cart-for-user-admin.yaml
+  - include:
+      file: db/changelog/changes/012-create-cart-items-table.yaml
```
Comment (okuzan) [inline src/main/resources/db/changelog/db.changelog-master.yaml:25]:
```suggestion
      file: db/changelog/changes/012-create-cart-items-table.yaml

```

File: src/main/java/book/store/onlinebookstore/service/CartItemService.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package book.store.onlinebookstore.service;
+
+import book.store.onlinebookstore.dto.cartitemdto.CartItemRequestDto;
+import book.store.onlinebookstore.dto.cartitemdto.CartItemUpdateRequestDto;
+import java.security.Principal;
+
+public interface CartItemService {
+
```
Comment (okuzan) [inline src/main/java/book/store/onlinebookstore/service/CartItemService.java:8]:
```suggestion
```

**Review by okuzan — CHANGES_REQUESTED** [review]
remove redundancy in subpackages, dto/categorydto to dto/category etc

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**